### PR TITLE
【1.1.15】Spark engine add conf 

### PR DESCRIPTION
**Pull request by v-kkhuang**
### What is the purpose of the change

spark  add  spark config：
1. spark.conf 
2. spark.locality.wait 
3. spark.memory.fraction

### Related issues/PRs

Related spark.conf  issues: #4880
Related spark.locality.wait/spark.memory.fraction   issues: #4881

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-dist/package/db/linkis_dml.sql [hunk 1]
```diff
@@ -89,13 +89,14 @@ INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`,
...
 INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`, `default_value`, `validate_type`, `validate_range`, `is_hidden`, `is_advanced`, `level`, `treeName`, `engine_conn_type`) VALUES ('wds.linkis.engineconn.max.free.time', '取值范围：3m,15m,30m,1h,2h', '引擎空闲退出时间','1h', 'OFT', '[\"1h\",\"2h\",\"30m\",\"15m\",\"3m\"]', '0', '0', '1', 'hive引擎设置', 'hive');
-
+INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`, `default_value`, `validate_type`, `validate_range`, `engine_conn_type`, `is_hidden`, `is_advanced`, `level`, `treeName`, `boundary_type`, `en_treeName`, `en_description`, `en_name`) VALUES ('spark.conf', 'spark自定义参数配置,多个参数使用分号;分隔', 'spark自定义配置参数',null, 'None', NULL, 'spark',0, 1, 1,'spark资源设置', 0, 'Spark resource settings','Spark custom parameter configuration, using semicolons for multiple parameters; separate', 'Spark custom configuration parameters');
```
Comment (casionone) [inline linkis-dist/package/db/linkis_dml.sql:97]: 'spark自定义参数配置,多个参数使用分号;分隔' -> `多个参数使用分号[;]分隔 例如spark.sql.shuffle.partitions=10;`
'Spark custom parameter configuration, using semicolons for multiple parameters; separate' ->'Multiple parameters are separated by semicolons [;] For example, spark.sql.shuffle.partitions=10;'

File: linkis-dist/package/db/linkis_dml.sql [hunk 2]
```diff
@@ -89,13 +89,14 @@ INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`,
...
+INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`, `default_value`, `validate_type`, `validate_range`, `engine_conn_type`, `is_hidden`, `is_advanced`, `level`, `treeName`, `boundary_type`, `en_treeName`, `en_description`, `en_name`) VALUES ('spark.locality.wait', '范围：0-3000，单位：毫秒', '任务调度本地等待时间', '3000', 'Regex', '^(0|1000|2000|3000)$', 'spark', 0, 1, 1, 'spark资源设置', 0, 'Spark resource settings', 'Range: 0-3000, Unit: millisecond', 'Task scheduling local waiting time');
+INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`, `default_value`, `validate_type`, `validate_range`, `engine_conn_type`, `is_hidden`, `is_advanced`, `level`, `treeName`, `boundary_type`, `en_treeName`, `en_description`, `en_name`) VALUES ('spark.memory.fraction', '范围：0.4,0.5,0.6，单位：百分比', 'JVM堆内存中M的百分比', '0.6', 'Regex', '^(0\\.4|0\\.5|0\\.6)$', 'spark', 0, 1, 1, 'spark资源设置', 0, 'Spark resource settings', 'Range: 0.4, 0.5, 0.6, in percentage', 'The percentage of M in JVM heap memory');
```
Comment (casionone) [inline linkis-dist/package/db/linkis_dml.sql:99]: JVM堆内存中M的百分比 -> 执行内存和存储内存的百分比

Comment (casionone) [reply linkis-dist/package/db/linkis_dml.sql:99]: 统一放到spark 参数一起

File: linkis-dist/package/db/linkis_dml.sql [hunk 3]
```diff
@@ -89,13 +89,14 @@ INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`,
 INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`, `default_value`, `validate_type`, `validate_range`, `is_hidden`, `is_advanced`, `level`, `treeName`, `engine_conn_type`) VALUES ('spark.tispark.tidb.port', NULL, NULL, '4000', 'None', NULL, '0', '0', '1', 'tidb设置', 'spark');
 INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`, `default_value`, `validate_type`, `validate_range`, `is_hidden`, `is_advanced`, `level`, `treeName`, `engine_conn_type`) VALUES ('spark.tispark.tidb.user', NULL, NULL, 'root', 'None', NULL, '0', '0', '1', 'tidb设置', 'spark');
 INSERT INTO `linkis_ps_configuration_config_key` (`key`, `description`, `name`, `default_value`, `validate_type`, `validate_range`, `is_hidden`, `is_advanced`, `level`, `treeName`, `engine_conn_type`) VALUES ('spark.python.version', '取值范围：python2,python3', 'python版本','python2', 'OFT', '[\"python3\",\"python2\"]', '0', '0', '1', 'spark引擎设置', 'spark');
--- hive
```
Comment (casionone) [inline linkis-dist/package/db/linkis_dml.sql:92]: 不需要移除

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/launch/SparkSubmitProcessEngineConnLaunchBuilder.scala [hunk 4]
```diff
@@ -62,6 +63,15 @@ class SparkSubmitProcessEngineConnLaunchBuilder(builder: JavaProcessEngineConnLa
     val executorCores = getValueAndRemove(properties, LINKIS_SPARK_EXECUTOR_CORES)
     val executorMemory = getValueAndRemove(properties, LINKIS_SPARK_EXECUTOR_MEMORY)
     val numExecutors = getValueAndRemove(properties, LINKIS_SPARK_EXECUTOR_INSTANCES)
+    val sparkcsonf = getValueAndRemove(properties, SPARK_CONF)
+    if (StringUtils.isNotBlank(sparkcsonf)) {
+      val strArrary = sparkcsonf.split(";").toList
+      strArrary.foreach { keyAndValue =>
+        val key = keyAndValue.split("=")(0)
```
Comment (casionone) [inline linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/launch/SparkSubmitProcessEngineConnLaunchBuilder.scala:70]: 加判定  keyAndValue.split("=") 可能为空  
为空的加warn日志

Comment (casionone) [reply linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/launch/SparkSubmitProcessEngineConnLaunchBuilder.scala:70]: 加示例注解

File: linkis-dist/package/db/linkis_dml.sql (hunk 1 above)
Comment (v-kkhuang) [reply linkis-dist/package/db/linkis_dml.sql:97]: done

File: linkis-dist/package/db/linkis_dml.sql (hunk 3 above)
Comment (v-kkhuang) [reply linkis-dist/package/db/linkis_dml.sql:92]: done

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/launch/SparkSubmitProcessEngineConnLaunchBuilder.scala (hunk 4 above)
Comment (v-kkhuang) [reply linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/launch/SparkSubmitProcessEngineConnLaunchBuilder.scala:70]: done

File: linkis-dist/package/db/linkis_dml.sql (hunk 2 above)
Comment (v-kkhuang) [reply linkis-dist/package/db/linkis_dml.sql:99]: done

**Review by casionone — APPROVED** [review]

**Pull request merged**